core/src/banking_stage/consumer.rs

```diff
@@ -392,14 +392,54 @@ impl Consumer {
         bank: &Arc<Bank>,
         txs: &[SanitizedTransaction],
         chunk_offset: usize,
+    ) -> ProcessTransactionBatchOutput {
+        // No filtering before QoS - transactions should have been sanitized immediately prior to this call
+        let pre_results = std::iter::repeat(Ok(()));
+        self.process_and_record_transactions_with_pre_results(bank, txs, chunk_offset, pre_results)
+    }
+
+    pub fn process_and_record_aged_transactions(
+        &self,
+        bank: &Arc<Bank>,
+        txs: &[SanitizedTransaction],
+        max_slot_ages: &mut [Slot],
+    ) -> ProcessTransactionBatchOutput {
+        // Need to filter out transactions since they were sanitized earlier.
+        // This means that the transaction may cross and epoch boundary (not allowed),
+        //  or account lookup tables may have been closed.
+        let pre_results = txs
+            .iter()
+            .zip(max_slot_ages)
+            .map(|(tx, max_slot_age)| {
+                if *max_slot_age < bank.slot() {
+                    *max_slot_age = 0; // Sentinal value to indicate transaction was sanitized before epoch boundary
+                    Err(TransactionError::BlockhashNotFound)
+                } else {
+                    let lookup_tables = tx.message().message_address_table_lookups();
+                    if !lookup_tables.is_empty() {
+                        bank.load_addresses(tx.message().message_address_table_lookups())?;
```

Comment:
nit: no need to call `.message_address_table_lookups()` twice? maybe we can consume `lookup_tables` above by passing to it?

Reply:
yeah, thanks for the catch: 04870fe31ada423e6943030254492e03c38c8d85